datajunction-server/datajunction_server/api/nodes.py

```diff
@@ -1301,6 +1304,45 @@ def create_a_cube(
     return node  # type: ignore
 
 
+@router.post(
+    "/register/table/{catalog}/{schema_}/{table}/",
+    response_model=NodeOutput,
+    status_code=201,
+)
+def register_a_table(  # pylint: disable=too-many-arguments
+    catalog: str,
+    schema_: str,
+    table: str,
+    session: Session = Depends(get_session),
+    settings: Settings = Depends(get_settings),
+    query_service_client: QueryServiceClient = Depends(get_query_service_client),
+) -> NodeOutput:
+    """
+    Register a table. This creates a source node in the SOURCE_NODE_NAMESPACE and
+    the source node's schema will be inferred using the configured query service.
+    """
+    if not query_service_client:
+        raise DJException(
+            message="Registering tables requires that a query "
+            "service is configured for table columns inference",
+        )
+    namespace = f"{settings.source_node_namespace}.{catalog}.{schema_}"
+    name = f"{namespace}.{table}"
+    create_a_node_namespace(namespace=namespace, session=session)
+    return create_a_source(
+        data=CreateSourceNode(
+            catalog=catalog,
+            schema_=schema_,
+            table=table,
+            name=name,
```

Comment:
Should the "display name" also be explicitly set to `f"{catalog}.{schema_}.{table}"`?

Reply:
I like that, it'll make things a lot more clear. I'll update this

Reply:
Working on this now, would that be misleading since users have to use the source node namespace prefix? In other words, should display name be the same as name--`f"{settings.source_node_namespace}.{catalog}.{schema_}.{table}"`?

Reply:
Yes, I think that the default display name should basically be the full node name. But I am not sure if we need to populate it in the DB or just pull it in the UI from the name, when display_name is empty.

Reply:
Yeah +1 to the display name being the full node name. I think it's better for the database to store the display name, since the overhead is quite small and otherwise we'll rely too much on that logic being there in the UI.

Reply:
Ok I updated this so the name and display will be the same for nodes created through the register table endpoint.